packages/aws-cdk-lib/aws-lambda-nodejs/lib/function.ts

```diff
@@ -101,27 +116,45 @@ export class NodejsFunction extends lambda.Function {
       throw new Error('Only `NODEJS` runtimes are supported.');
     }
 
-    // Entry and defaults
-    const entry = path.resolve(findEntry(id, props.entry));
-    const handler = props.handler ?? 'handler';
-    const architecture = props.architecture ?? Architecture.X86_64;
-    const depsLockFilePath = findLockFile(props.depsLockFilePath);
-    const projectRoot = props.projectRoot ?? path.dirname(depsLockFilePath);
     const runtime = getRuntime(scope, props);
 
-    super(scope, id, {
-      ...props,
-      runtime,
-      code: Bundling.bundle(scope, {
-        ...props.bundling ?? {},
-        entry,
+    if (props.code !== undefined) {
+      if (props.handler === undefined) {
+        throw new Error(
+          'Cannot determine handler when `code` property is specified. Use `handler` property to specify a handler.\n'
+          + 'The handler should be the name of the exported function to be invoked and the file containing that function.\n'
+          + 'For example, handler should be specified in the form `myFile.myFunction`',
+        );
+      }
+
+      super(scope, id, {
+        ...props,
+        runtime,
+        code: props.code,
+        handler: props.handler!,
```

Comment:
Can we remove the `!`? We're already checking and throwing an error if `props.handler` is undefined so we don't need to use `!` here.